base/precompilation.jl

```diff
@@ -919,10 +920,16 @@ function precompilepkgs(pkgs::Vector{String}=String[];
     seconds_elapsed = round(Int, (time_ns() - time_start) / 1e9)
     ndeps = count(values(was_recompiled))
     if ndeps > 0 || !isempty(failed_deps) || (quick_exit && !isempty(std_outputs))
-        str = sprint() do iostr
+        str = sprint(context=io) do iostr
             if !quick_exit
+                if fancyprint # replace the progress bar
+                    packages = isempty(requested_pkgs) ? "packages." : "$(join(requested_pkgs, ", "))."
+                    printpkgstyle(iostr, :Precompiling, packages, newline=false)
```

Comment:
@KristofferC what about
```suggestion
                    printpkgstyle(iostr, :Precompiling, "finished. " * packages, newline=false)
```